plugins/inputs/net_response/net_response.go

```diff
@@ -114,11 +122,16 @@ func (n *NetResponse) UdpGather() (map[string]interface{}, error) {
 	LocalAddr, err := net.ResolveUDPAddr("udp", "127.0.0.1:0")
 	// Connecting
 	conn, err := net.DialUDP("udp", LocalAddr, udpAddr)
-	defer conn.Close()
 	// Handle error
 	if err != nil {
-		return nil, err
+		if e, ok := err.(net.Error); ok && e.Timeout() {
+			fields["result_type"] = "timeout"
+		} else {
+			fields["result_type"] = "connection_failed"
+		}
+		return fields, nil
```

Comment:
I think it would be a good starting point to just have `connection_failed` here, we can always add more specific states later if needed.